### PR TITLE
Update chats endpoints tests

### PR DESCRIPTION
**Pull request by nedjulius**
Solves #181 

Updated unit tests for chats routes.

### CODE REVIEW

File: back-end/routes/chats.test.js [hunk 1]
```diff
@@ -1,74 +1,157 @@
...
+    const {id} = await User.create({
+      username: authUsername,
+      password: authPassword,
+    });
+    const res = await request
+      .post('/auth/login')
+      .send({username: authUsername, password: authPassword});
+
+    authUserId = id;
+    authHeader = {Authorization: `Bearer ${res.body.token}`};
```
Comment (mnu213) [inline back-end/routes/chats.test.js:25]: Is this the fix for the needing a jwt token? If so I am confused how this is being sent in the res.body.token if this is just a mock?

**Review by mnu213 — APPROVED** [review]

File: back-end/routes/chats.test.js [hunk 2]
```diff
@@ -1,74 +1,157 @@
...
+      const secondChat = await Chat.create({
+        name: secondChatTitle,
+        users: [authUserId],
+      });
+      await User.updateOne(
+        {_id: authUserId},
+        {chats: [firstChat.id, secondChat.id]}
+      );
+
+      const response = await request.get('/chats').set(authHeader);
```
Comment (mnu213) [inline back-end/routes/chats.test.js:53]: Related to above comment, what is this that you do in every test?

File: back-end/routes/chats.test.js (hunk 1 above)
Comment (nedjulius) [reply back-end/routes/chats.test.js:25]: it's not a mock, it's just a regular token. I import the whole app and send a login request to the express app via supertest module `request.post('/login').send({username, password})`. then I just get the token from the server and attach it to every request that requires authentication.

File: back-end/routes/chats.test.js (hunk 2 above)
Comment (nedjulius) [reply back-end/routes/chats.test.js:53]: this is how i attach the authorization headers with the auth token.

**Pull request merged**